### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by jamieRollison**
Added more tests to see how BFS and Dijkstra's handles cycles, and if they agree with each other on the real dataset. Also added input validation on dijkstra's and BFS to ensure both the start and end are valid vertices.

### CODE REVIEW

File: src/wikigraph.cc [hunk 1]
```diff
@@ -158,4 +167,8 @@ std::vector<WikiGraph::RankedPage> WikiGraph::rankPages() const {
   std::vector<RankedPage> ranked_pages;
   // ...
   return ranked_pages;
+}
+
+bool WikiGraph::validStartAndEnd(const std::string& start_page, const std::string& end_page) const {
```
Comment (opcushing) [inline src/wikigraph.cc:166]: Ah I like this change. Good validation.

File: tests/tests.cc [hunk 2]
```diff
@@ -72,4 +84,22 @@ TEST_CASE("BFS picks the best path", "BFS") {
...
+  };
+  WikiGraph w("./datasets/cycle.tsv");
+  REQUIRE(expectedAE == w.getPathBFS("A", "E"));
+  REQUIRE(expectedAD == w.getPathBFS("A", "D"));
+}
+
+TEST_CASE("BFS matches Dijkstra's given real data", "real") {
+  WikiGraph w("./datasets/links.tsv");
+
+  REQUIRE(w.getPathBFS("12th century", "Walrus") == w.getPathDijkstras("12th century", "Walrus"));
```
Comment (opcushing) [inline tests/tests.cc]: Good so far. I think we can add some more testing on the real data set. We'll have to be a bit more thorough for the demo.

**Review by opcushing — CHANGES_REQUESTED** [review]
Good changes. I do think we can add some more thorough testing, however. I think we should add more real dataset tests, and test more edge cases when it comes to validation, such as invalid start / end paths, and cases where there is no valid path.

**Review by opcushing — APPROVED** [review]
Looks great. Thanks.
